### PR TITLE
Post Deletion

### LINKED ISSUE #90: [Post view] [FE] Post deletion
## Context

#124 took care of the backend work for post deletion. We want to enable this functionality by building out the frontend for this feature. See #91 for additional context on why admins would want to delete certain posts.

## Requirements

Reference the Figma designs to build out the frontend. Utilize the API route `post.delete` we have to mutate the post. It is expected that you implement the button that opens the deletion confirmation as well - make sure that this button is only shown to admins and content creators, and not volunteers. After this ticket is complete, we should be able to go through the entire lifecycle of a post (except for marking as covered/hiding).

If an error occurs on the backend, display a toast notifying the user that the deletion was unsuccessful.

#### PM

- [ ] Deleting a post should successfully remove it from the feed. The post cannot be recovered again, and cannot be viewed anywhere on the platform.

#### EM

- [ ] Toast appears on errors (should not display a toast when there the error is on the storage bucket side)
- [ ] Deleting a post takes the user back to the feed (closes all modals)

#### Designer

- [ ] Make sure the delete confirmation opens a modal

### PR DESCRIPTION
**Pull request by joannacheng21**
## Post Deletion

Issue Number(s): #90 

This PR adds the frontend button + modal for post deletion and uses the API route to delete the post on the backend. Displays a toast if there's an error deleting the post.

### Checklist

- [x] Requirements have been implemented
- [x] Acceptance criteria are met
- [x] Database schema docs have been updated or are not necessary
- [x] Code follows design and style guidelines
- [x] Commits follow guidelines (concise, squashed, etc)
- [x] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- None

### Related PRs

- None

### Testing

Try putting in an invalid post id to check the toast. Delete posts and make sure they don't show up in post anymore (invalidate post router so posts are refetched and deleted post stops showing up).

### CODE REVIEW

File: web/components/Feed/Feed.tsx
Comment (katsukixyz) [inline web/components/Feed/Feed.tsx]: Use `Types.ObjectId` instead

File: web/components/PetPostModal/PetPostModal.tsx
Comment (katsukixyz) [inline web/components/PetPostModal/PetPostModal.tsx]: See previous comment

File: web/components/PetPostModal/PetPostModal.tsx
Comment (katsukixyz) [inline web/components/PetPostModal/PetPostModal.tsx]: Assuming this was used for testing - could we remove this without affecting functionality?

File: web/components/PetPostModal/PetPostModal.tsx
Comment (katsukixyz) [inline web/components/PetPostModal/PetPostModal.tsx]: Use `text-secondary` color

File: web/components/PetPostModal/PetPostModal.tsx
Comment (katsukixyz) [inline web/components/PetPostModal/PetPostModal.tsx]: Is this reload necessary if we invalidate?

File: web/components/PetPostModal/PetPostModal.tsx
Comment (katsukixyz) [inline web/components/PetPostModal/PetPostModal.tsx]: Consider pulling this out as its own component

File: web/components/PetPostModal/PetPostModal.tsx
Comment (katsukixyz) [inline web/components/PetPostModal/PetPostModal.tsx]: Use the `leftIcon` property built into Chakra - take a look at https://chakra-ui.com/docs/components/button#button-with-icon

**Review by katsukixyz — COMMENTED** [review]
Looking good! Make sure to merge in staging when you get a chance.

File: web/components/PetPostModal/DeleteButtonModal.tsx
Comment (katsukixyz) [inline web/components/PetPostModal/DeleteButtonModal.tsx]: Move this back into the Post modal so that the deletion modal component only contains the modal itself

File: web/components/PetPostModal/DeleteButtonModal.tsx
Comment (katsukixyz) [inline web/components/PetPostModal/DeleteButtonModal.tsx]: Rename to `DeletePostModal`

**Review by katsukixyz — COMMENTED** [review]
Minor changes! I'll ping the LT to get their feedback!

**Comment by rohan-chadha** [comment]
Looks great! Nothing from me!

**Review by Wesley257 — APPROVED** [review]

**Review by katsukixyz — APPROVED** [review]
LGTM! Good work. Once you have an approval from @winniezhangg, feel free to merge it in!

**Review by winniezhangg — APPROVED** [review]
Looks good Joanna! Great work!

**Pull request merged**
